browser-extension/mv2/src/background/background.js

```diff
@@ -974,6 +974,61 @@ BG.Methods.sendMessage = function (messageObject, callback) {
   );
 };
 
+BG.Methods.sendMessageToApp = (messageObject, tabId, timeout = 2000) => {
+  const sendMessage = (messageObject, tabId) => {
```

Comment:
This function will not get the message object from anywhere

As discussed, this is not a reusable function for now and will also be responsible for reading events from local storage

so you still have to create a separate function called `sendEventsToTab` that does all this and also the batching of events to handle the edge cases / acknowledgement procedure of this function

Reply:
No, we anyway need a `sendMessageToApp` which will be a general, reusable function. Whatever event function we create, internally in the last step, it must call `sendMessageToApp` to send the event to App/UI.